src/api/block_api.py

```diff
@@ -11,12 +10,15 @@ def __init__(self, nw):
     def get_current_level(self):
         pass
 
-    def level_to_cycle(self, level):
-        return floor(level / self.nw['BLOCKS_PER_CYCLE'])
-
+    @abstractmethod
     def get_current_cycle(self):
-        return self.level_to_cycle(self.get_current_level())
+        pass
 
     def level_in_cycle(self, level):
-        cycle = self.level_to_cycle(level)
-        return level - (cycle * self.nw['BLOCKS_PER_CYCLE']) - 1
+        level = self.get_current_level()
+        if level > 1589248:
+            # After Granada
```

Comment:
Is this check running only on mainnet?

Reply:
no, but I see how this is a problem as the test network starts at block 1 with one protocol (or quick transition). We should put the numbers in a constant at the top of the file and check here for mainnet or testnet. Can you fix it?

Reply:
Constants should be defined at the top of the file in all CAPS, or imported from a constants provider file. And yes, this value should be dependent on mainnet/testnet flag.

Reply:
I agree, but:

- tests do not seem to be using this function
- before this PR, it was still "problematic", given the fact that mainnet and testnet were always different
- we have `1589248 * 15` seconds (=275 days) until this becomes a problem on Granadanet

Reply:
actually granadanet started on florence and updated to granada at block 8096. but yes, we can fix later, this will work as is. I won't be able to test the PR until tomorrow.

Reply:
@denver-s at least move the magic numbers to constants,please

Reply:
@jdsika done https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/474/commits/d2278c84484ff207e9d3c43fc1c81c129a177b1c